### PR TITLE
Feature/change orgin URL for subkey generation

### LINKED ISSUE #2446: Provide a way to locally regenerate a subkey for a different domain
## Description
Subkey (Raiden Account, #172) is a unique private key used in the Light Client, deterministically generated as the `sha256` of the main key `personal_sign` (EIP-191)  signature of a message containing the current network name, dApp's domain and a warning that this message should not be signed outside of a trusted app, or a different domain, or else it'd give the webapp access to its funds and Raiden channels.

We need to make it more secure, and also make it easier for a user to regenerate this key for another domain, to control the channels and funds from a standalone, locally served instance of the dApp, in case that domain goes offline.

This should be done before mainnet release, as it's a breaking change.
Two proposals:
1. Read an environment variable to allow the user to specify the domain/origin/referer to be used in the subkey message to be signed
2. Investigate if EIP-712 standard provide more security for our usecase, by embedding the domain in the payload, and having Metamask enforce it. This may only be feasible if there're advanced options allowing 1. above to still be possible, although with the proper safeguards on Metamask's side.

## Acceptance criteria
- We do not use EIP-712
- The user can set an environment variable
- If the environment variable is defined it is overriding the URL
- The workaround is documented

## Tasks
- [ ]

### PR DESCRIPTION
**Pull request by weilbith**
Fixes #2446

**Short description**
Details of the design decision in the commit messages. Else a pretty simple and small PR...

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. Add the `VUE_APP_SUBKEY_ORIGIN_URL` variable to the `.env.development` file and assign any random value to it.
2. Serve the dApp locally
3. Access the dApp in the web-browser and attempt to connect
4. Verify that the message that you should sign for the subkey generation includes the URL you have specified before

### CODE REVIEW

File: raiden-ts/docs-source/installing-dapp.md [hunk 1]
```diff
@@ -24,3 +24,24 @@ yarn workspace raiden-dapp run serve
 Once the development server has been started, navigate to `http://localhost:8080` to start using the Raiden dApp.
 
 > Note that for the Raiden dApp to work it requires MetaMask or some other Web3 provider (wallet apps with dApp support) to be installed on your browser.
+
+## Recover Backup locally
+
+You might have a backup by a dApp instance that was provided by a service that
+is not available anymore. You can not import this backup to another dApp
+provider if you was using a Raiden subkey (configuration option). The subkey
```
Comment (andrevmatos) [inline raiden-ts/docs-source/installing-dapp.md:32]: Were

File: raiden-ts/src/helpers.ts [hunk 2]
```diff
@@ -95,10 +95,11 @@ export const getContracts = (network: Network): ContractsInfo => {
  *
  * @param network - Network to include in message
  * @param main - Main signer to derive subkey from
+ * @param origin_url - URL of the origin to generate the subkey for
  * @returns Subkey's signer & address
  */
-async function genSubkey(network: Network, main: Signer) {
-  const url = globalThis?.location?.origin ?? 'unknown';
+async function genSubkey(network: Network, main: Signer, origin_url?: string) {
```
Comment (andrevmatos) [inline raiden-ts/src/helpers.ts:101]: Please, use camelCase only for variables and parameters. Snake_case is allowed only for API objects or uppercased constants

File: raiden-ts/src/helpers.ts [hunk 3]
```diff
@@ -121,12 +122,14 @@ Signing this message at any other address WILL give it FULL control of this subk
...
  * @param provider - a provider
  * @param subkey - Whether to generate a subkey
+ * @param subkey_origin_url - URL of the origin to generate a subkey for
  * @returns a [[Signer]] or [[Wallet]] that can be used for signing
  */
 export const getSigner = async (
   account: string | number | Signer,
   provider: JsonRpcProvider,
   subkey?: boolean,
+  subkey_origin_url?: string,
```
Comment (andrevmatos) [inline raiden-ts/src/helpers.ts:132]: Same

File: raiden-ts/src/raiden.ts [hunk 4]
```diff
@@ -396,6 +398,7 @@ export class Raiden {
     contractsOrUDCAddress?: ContractsInfo | string,
     config?: Decodable<PartialRaidenConfig>,
     subkey?: true,
+    subkey_origin_url?: string,
```
Comment (andrevmatos) [inline raiden-ts/src/raiden.ts:401]: Same

**Review by andrevmatos — APPROVED** [review]
Just a few comments, merge after they're addressed. Thanks

File: raiden-ts/docs-source/installing-dapp.md (hunk 1 above)
Comment (weilbith) [reply raiden-ts/docs-source/installing-dapp.md:32]: Oh damn, yes of course! Wrote too much Python privately. :see_no_evil:

File: raiden-ts/src/helpers.ts (hunk 2 above)
Comment (weilbith) [reply raiden-ts/src/helpers.ts:101]: Oh damn, yes of course! I wrote too much Python privately the last days. :see_no_evil: 
But why does linting not report? :scream:

File: raiden-ts/docs-source/installing-dapp.md (hunk 1 above)
Comment (weilbith) [reply raiden-ts/docs-source/installing-dapp.md]: :heavy_check_mark:

File: raiden-ts/src/helpers.ts (hunk 2 above)
Comment (weilbith) [reply raiden-ts/src/helpers.ts]: :heavy_check_mark:

File: raiden-ts/src/helpers.ts (hunk 3 above)
Comment (weilbith) [reply raiden-ts/src/helpers.ts]: :heavy_check_mark:

File: raiden-ts/src/raiden.ts (hunk 4 above)
Comment (weilbith) [reply raiden-ts/src/raiden.ts]: :heavy_check_mark:

**Comment by weilbith** [comment]
Aright, review feedback got targeted and CI runs through. I'm rebasing now to merge the fixup commits.

**Pull request merged**